src/main/kotlin/no/nav/fo/forenkletdeploy/consumer/GithubConsumer.kt

```diff
@@ -0,0 +1,143 @@
+package no.nav.fo.forenkletdeploy.consumer
+
+import no.nav.fo.forenkletdeploy.ApplicationConfig
+import no.nav.fo.forenkletdeploy.util.Utils
+import org.slf4j.LoggerFactory
+import org.springframework.cache.annotation.Cacheable
+import org.springframework.context.annotation.Profile
+import org.springframework.stereotype.Service
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+import java.util.*
+
+@Service("GithubConsumer")
+@Profile("!mock")
+open class GithubConsumer : StashConsumer {
+    val LOG = LoggerFactory.getLogger(this.javaClass)
+    val LIMIT = 1000
+    val TOKEN = Utils.getRequiredProperty("GITHUB_JENKINSPUS_TOKEN")
+
+    @Cacheable("githubcommits")
+    override fun getCommits(application: ApplicationConfig, fromTag: String, toTag: String): List<StashCommit> =
+            try {
+                val useFromTag = if (fromTag == null || fromTag == "null") "1" else "${fromTag}"
+                val useToTag = if (toTag == null || toTag == "null") "HEAD" else "${toTag}"
+
+                val url = "${getRestUriForGithubRepo(application)}/compare/${useFromTag}...${useToTag}"
+                LOG.info("Henter commits for ${application.name} ($fromTag -> $toTag) via $url")
+                Utils.withClient(url)
+                        .request()
+                        .header("Authorization", "token ${TOKEN}")
+                        .get(GithubCommits::class.java)
+                        .commits
+                        .map { StashCommit(
+                                id = it.sha,
+                                author = CommitPerson(true, 1,
```

Comment:
Jeg er personlig veldig glad i at man bruker named-parameters i kotlin. Rett og slett fordi jeg mener det gjør koden mer robust for refakturering. Om man endrer en dataklasse, eksempelvis legger til flere felter eller endrer rekkefølgen - så vil man med named parameters ikke få feil. Mens uten vil du kun få feil om typene på argumentene faktisk endrer seg.

Jeg har kanskje ikke vært så konsekvent på det selv, i hvertfall ikke for funksjoner med kun 1. parameter. Men når man får flere parameter inn tror jeg det gir en del verdi :)

Men åpen for andre innspill og synspunkter :)